### PR TITLE
Measurements

### PR DESCRIPTION
**Pull request by Andreas-Kreouzos**
Completed the addition of the Measurements Entity, as well as the writing of tests only the service class at the moment. Done minor refactoring to the whole project

### CODE REVIEW

File: src/main/java/com/project/trackfit/core/ApplicationUserController.java [hunk 1]
```diff
@@ -22,13 +21,13 @@
...
+
     @PostMapping
-    public ResponseEntity<APICustomResponse>createUser(
-            @Valid
-            @RequestBody
-            CreateUserRequest createUserRequest) throws NoSuchAlgorithmException {
-        UUID userId=applicationUserService.createUser(createUserRequest);
+    public ResponseEntity<APICustomResponse> createUser(@Valid @RequestBody CreateUserRequest createUserRequest)
+            throws NoSuchAlgorithmException {
+        UUID userId = applicationUserService.createUser(createUserRequest);
```
Comment (amigoscode) [inline src/main/java/com/project/trackfit/core/ApplicationUserController.java]: why NoSuchAlgorithmException

File: src/main/java/com/project/trackfit/core/CreateUserRequest.java [hunk 2]
```diff
@@ -5,6 +5,7 @@ public record CreateUserRequest(
         String password,
         String firstName,
         String lastName,
-        String  role
-) {
-}
+        String role,
```
Comment (amigoscode) [inline src/main/java/com/project/trackfit/core/CreateUserRequest.java]: ROLE should be enum

**Review by ghost — DISMISSED** [review]

File: src/main/java/com/project/trackfit/core/ApplicationUserController.java (hunk 1 above)
Comment (Andreas-Kreouzos) [reply src/main/java/com/project/trackfit/core/ApplicationUserController.java]: This exception has to be handled when using the following code snippet: ``md = MessageDigest.getInstance("SHA-512");``

To resolve this issue we added a try / catch block and removed the ``throws NoSuchAlgorithmException`` from every other method declaration.

File: src/main/java/com/project/trackfit/core/CreateUserRequest.java (hunk 2 above)
Comment (Andreas-Kreouzos) [reply src/main/java/com/project/trackfit/core/CreateUserRequest.java]: Done!!!

**Review by ghost — APPROVED** [review]

**Pull request merged**